File: code_samples/scala/data_structures_heap.scala
/**
 * Data Structures: Heap
 * AI/ML Training Sample
 */

class Heap {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object HeapExample {
  def main(args: Array[String]): Unit = {
    val instance = new Heap()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/lua/data_structures_heap.lua
-- Data Structures: Heap
-- AI/ML Training Sample

Heap = {}
Heap.__ index = Heap

function Heap:new()
    local obj = {
        data = ""
    }
    setmetatable(obj, self)
    return obj
end

function Heap:process(input)
    self.data = input
end

function Heap:getData()
    return self.data
end

function Heap:validate()
    return #self.data > 0
end

-- Example usage
local instance = Heap:new()
instance:process("example")
print("Data: " .. instance:getData())
print("Valid: " .. tostring(instance:validate()))
